### PR TITLE
support AveragePool19 and Pad19

### PR DESCRIPTION
**Pull request by liqunfu**
### Description
[](https://github.com/microsoft/onnxruntime/issues/15569)

### Motivation and Context
AveragePoolXDTask can be refactored with LpPoolXDTask. However, there has been an issue with LpPool in ONNX (https://github.com/microsoft/onnxruntime/issues/6302). with the issue fixed, LpPoolXDTask can be implemented using AveragePoolXDTask. This is not in the scope of this PR.

### CODE REVIEW

**Review by xadupre — APPROVED** [review]

File: onnxruntime/core/providers/cpu/nn/pool.cc [hunk 1]
```diff
@@ -249,6 +249,81 @@ Status MaxPoolV8::ComputeImpl(OpKernelContext* context) const {
...
+      const int64_t dilation_d = pool_attrs_.dilations[2];
+      RunLoop<AveragePool3DTask<T>>(tp, onnxruntime::narrow<size_t>(total_channels),
+                               {X_data, Y_data, x_step, y_step,
+                                dilation_h, dilation_w, dilation_d, pooled_height, pooled_width,
+                                pooled_depth, stride_h(), stride_w(), stride_d(), height,
+                                width, depth, kernel_shape, pads, p_});
+      break;
+    }
+    default:
+      return Status(ONNXRUNTIME, INVALID_ARGUMENT, "Unsupported kernel dimension : " + std::to_string(kernel_shape.size()));
```
Comment (justinchuby) [inline onnxruntime/core/providers/cpu/nn/pool.cc:321]: Also include all the supported ones to be for helpful for users?

**Comment by gramalingam** [comment]
Nit: update the title and description to indicate that this PR also covers "Pad-19" ?

File: onnxruntime/core/providers/cpu/tensor/pad.cc [hunk 2]
```diff
@@ -438,12 +464,46 @@ static Status PadImpl(OpKernelContext* ctx,
           int64_t prePad = reshaped_pad[inner_axis];
           int64_t postPad = reshaped_pad[inner_axis + new_dims_count];
           if (inner_no_pad_size == 1) {
-            PadInnermostAxis(axisStart - prePad, axisStart + prePad, -1 /* inputDelta */, onnxruntime::narrow<size_t>(prePad));
-            PadInnermostAxis(output, output - 2, -1 /* inputDelta */, onnxruntime::narrow<size_t>(postPad));
+            if (mode == Mode::Reflect) {
+              PadInnermostAxis(axisStart - prePad, axisStart + prePad, -1 /* inputDelta */, onnxruntime::narrow<size_t>(prePad));
+              PadInnermostAxis(output, output - 2, -1 /* inputDelta */, onnxruntime::narrow<size_t>(postPad));
+            } else {
```
Comment (gramalingam) [inline onnxruntime/core/providers/cpu/tensor/pad.cc:470]: Nit: a comment like `// mode == Mode::Wrap` would be helpful in the else-line (here and similarly below).

**Review by gramalingam — APPROVED** [review]

**Comment by liqunfu** [comment]
Pad19: https://github.com/microsoft/onnxruntime/issues/15576
AveragePool19: https://github.com/microsoft/onnxruntime/issues/15569

**Pull request merged**